### DISCUSSION TITLE
DropDown with content portal?

### DISCUSSION THREAD

**Discussion by Spiralis** [saadeghi/daisyui#1667] [Q&A] [answered]
Not sure if this is possible, but I wish that I could separate the content from the trigger, like you can with the modal. As in being able to render a dropdown in the context of the document, and not the node that the trigger is in. That node can have limited space, so overflows will occur when rendering it.

You can render the whole thing outside, i.e. in the document, body or similar, and then just visually hiode the button and manipulate a click on it when it is supposed to show, but then all the positioning is gone, And the positioning is kinda key for a dropdown...

Any suggestions?

**Comment by taochu** [comment]
+1

**Comment by saadeghi** [comment, marked as answer]
It's not possible with CSS but I think what you're looking for is [Floating UI](https://floating-ui.com/) (formerly popper.js)  
This library can dynamically append elements to the `<body>` which fixes the overflow crop problem and can also automatically compute the position based on the empty space available.

here's an example with daisyUI's `btn` and `menu`:  
https://stackblitz.com/edit/daisyui-react-vite-floatingui?file=src%2FApp.jsx

```jsx
import { useState } from 'react';
import {
  useFloating,
  useInteractions,
  useClick,
  useFocus,
  autoPlacement,
  offset,
  autoUpdate,
} from '@floating-ui/react';

import './App.css';

function App() {
  const [isOpen, setIsOpen] = useState(false);

  const { x, y, strategy, refs, context } = useFloating({
    open: isOpen,
    onOpenChange: setIsOpen,
    middleware: [autoPlacement(), offset(4)],
    whileElementsMounted: autoUpdate,
  });

  const click = useClick(context);
  const focus = useFocus(context);

  const { getReferenceProps, getFloatingProps } = useInteractions([
    click,
    focus,
  ]);

  return (
    <div class="px-[25rem] py-[30rem]">
      <div class="w-24 p-2 rounded-box border border-base-content overflow-hidden">
        <button
          className="btn"
          ref={refs.setReference}
          {...getReferenceProps()}
        >
          Click
        </button>
        This div has overflow-hidden
      </div>
      {isOpen && (
        <ul
          className="menu bg-base-200 w-56 p-2 rounded-box"
          ref={refs.setFloating}
          {...getFloatingProps()}
          style={{
            position: strategy,
            top: y ?? 0,
            left: x ?? 0,
          }}
        >
          <li>
            <a>Item 1</a>
          </li>
          <li>
            <a>Item 2</a>
          </li>
          <li>
            <a>Item 3</a>
          </li>
        </ul>
      )}
    </div>
  );
}

export default App;

```
